aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,242 @@
+
+To create a handler, define a class deriving from `DelegatingHandler`. Override the `SendAsync` method to execute code before passing the request to the next handler in the pipeline:
+
+[!code-csharp[Main](http-requests/samples/Handlers/ValidateHeaderHandler.cs?name=snippet1)]
+
+The preceding code defines a basic handler. It checks to see if an X-API-KEY header has been included on the request. If the header is missing, it can avoid the HTTP call and return a suitable response.
+
+During registration, one or more handlers can be added to the configuration for a `HttpClient`. This task is accomplished via extension methods on the `IHttpClientBuilder`.
+
+[!code-csharp[](http-requests/samples/Startup.cs?name=snippet5)]
+
+In the preceding code, the `ValidateHeaderHandler` is registered with DI. The handler **must** be registered in DI as transient. Once registered, `AddHttpMessageHandler` can be called, passing in the type for the handler.
+
+Multiple handlers can be registered in the order that they should execute. Each handler wraps the next handler until the final `HttpClientHandler` executes the request:
+
+[!code-csharp[](http-requests/samples/Startup.cs?name=snippet6)]
+
+## Use Polly-based handlers
+
+`IHttpClientFactory` integrates with a popular third-party library called [Polly](https://github.com/App-vNext/Polly). Polly is a comprehensive resilience and transient fault-handling library for .NET. It allows developers to express policies such as Retry, Circuit Breaker, Timeout, Bulkhead Isolation, and Fallback in a fluent and thread-safe manner.
+
+Extension methods are provided to enable the use of Polly policies with configured `HttpClient` instances. The Polly extensions are available in a NuGet package called 'Microsoft.Extensions.Http.Polly'. This package is not included by default by the 'Microsoft.AspNetCore.App' metapackage. To use the extensions, a PackageReference should be explicitly included in the project.
+
+[!code-csharp[](http-requests/samples/HttpClientFactorySample.csproj?highlight=9)]
+
+After restoring this package, extension methods are available to support adding Polly-based handlers to clients.
+
+### Handle transient faults
+
+The most common faults you may expect to occur when making external HTTP calls are transient. A convenient extension method called `AddTransientHttpErrorPolicy` is included, which allows a policy to be defined. The policy applies when these common transient errors occur. Examples of errors handled by the policy include `HttpRequestException`, HTTP 5xx responses, and HTTP 408 responses.
```

Comment:
"Examples of errors ... include" may suggest there are other kinds of error not mentioned which are also included, but what is stated _is_ the full set.  Maybe avoid "includes"?

Could:
>Policies configured with this extension method handle `HttpRequestException`, HTTP 5xx responses, and HTTP 408 responses.